server/base/src/main/java/org/apache/accumulo/server/master/balancer/HostRegexTableLoadBalancer.java

```diff
@@ -0,0 +1,334 @@
+    return InetAddress.getByName(hostIp).getHostName();
+  }
+
+  /**
+   * Matches table name against pool names, returns matching pool name or DEFAULT_POOL.
+   *
+   * @param tableName
+   *          name of table
+   * @return tablet server pool name (table name or DEFAULT_POOL)
+   */
+  protected String getPoolNameForTable(String tableName) {
+    if (null == tableName) {
+      return DEFAULT_POOL;
+    }
+    return poolNameToRegexPattern.containsKey(tableName) ? tableName : DEFAULT_POOL;
+  }
+
+  /**
+   * Parse configuration and extract properties
+   *
+   * @param conf
+   *          server configuration
+   */
+  protected void parseConfiguration(ServerConfiguration conf) {
+    tableIdToTableName = new HashMap<>();
+    poolNameToRegexPattern = new HashMap<>();
+    for (Entry<String,String> table : getTableOperations().tableIdMap().entrySet()) {
+      tableIdToTableName.put(table.getValue(), table.getKey());
+      Map<String,String> customProps = conf.getTableConfiguration(table.getValue()).getAllPropertiesWithPrefix(Property.TABLE_ARBITRARY_PROP_PREFIX);
+      if (null != customProps && customProps.size() > 0) {
+        for (Entry<String,String> customProp : customProps.entrySet()) {
+          if (customProp.getKey().startsWith(HOST_BALANCER_PREFIX)) {
+            String tableName = customProp.getKey().substring(HOST_BALANCER_PREFIX.length());
+            String regex = customProp.getValue();
+            poolNameToRegexPattern.put(tableName, Pattern.compile(regex));
+          }
+        }
+      }
+    }
+    String oobProperty = conf.getConfiguration().get(HOST_BALANCER_OOB_CHECK);
+    if (null != oobProperty) {
+      oobCheckMillis = AccumuloConfiguration.getTimeInMillis(oobProperty);
+    }
+    String poolRecheckProperty = conf.getConfiguration().get(HOST_BALANCER_POOL_RECHECK_KEY);
+    if (null != poolRecheckProperty) {
+      poolRecheckMillis = AccumuloConfiguration.getTimeInMillis(poolRecheckProperty);
+    }
+    String ipBased = conf.getConfiguration().get(HOST_BALANCER_REGEX_USING_IPS);
+    if (null != ipBased) {
+      isIpBasedRegex = Boolean.parseBoolean(ipBased);
+    }
+  }
+
+  public Map<String,String> getTableIdToTableName() {
+    return tableIdToTableName;
+  }
+
+  public Map<String,Pattern> getPoolNameToRegexPattern() {
+    return poolNameToRegexPattern;
+  }
+
+  public long getOobCheckMillis() {
+    return oobCheckMillis;
+  }
+
+  public long getPoolRecheckMillis() {
+    return poolRecheckMillis;
+  }
+
+  public boolean isIpBasedRegex() {
+    return isIpBasedRegex;
+  }
+
+  @Override
+  public void init(ServerConfiguration conf) {
+    super.init(conf);
+    parseConfiguration(conf);
+  }
+
+  @Override
+  public void getAssignments(SortedMap<TServerInstance,TabletServerStatus> current, Map<KeyExtent,TServerInstance> unassigned,
+      Map<KeyExtent,TServerInstance> assignments) {
+
+    Map<String,SortedMap<TServerInstance,TabletServerStatus>> pools = splitCurrentByRegex(current);
```

Comment:
This seems like it would be problematic. I would have to wait a default of 5minutes for tablets to be migrated to a new server in a pool (because the pools would not be recomputed). Did I miss something?